tests/acceptance/guid-file/registration-file-detail-test.ts

```diff
@@ -98,7 +112,167 @@ module('Acceptance | guid file | registration files', hooks => {
+            'File metadata target title text is properly set.');
+        assert.dom('[data-test-target-title]').exists();
+        // Target description
+        assert.dom('[data-test-target-description-label]').hasText('Description',
+            'File metadata target description text is properly set.');
+        assert.dom('[data-test-target-description]').exists();
+    });
+
+    // Save and cancel buttons with write permissions
+    test('save and cancel', async function(this: ThisTestContext, assert) {
+        await visit(`/--file/${this.file.id}`);
+        assert.equal(currentURL(), `/--file/${this.file.guid}`);
+
+        // Verify download
+        if (!this.registration.isAnonymous) {
+            assert.dom('[data-test-download-button]').exists();
+        }
 
-    // test for switching views
+        // Verify edit
+        if (this.file.target.currentUserPermissions.includes(Permission.Write)) {
+            await click('[data-test-edit-metadata-button]');
+            assert.dom('[data-test-title-field]').exists();
+            assert.dom('[data-test-description-field]').exists();
+            assert.dom('[data-test-select-resource-type]').exists();
+            assert.dom('[data-test-select-resource-language]').exists();
+            assert.dom('[data-test-cancel-editing-metadata-button]').exists();
+            assert.dom('[data-test-save-metadata-button]').exists();
+            await click('[data-test-cancel-editing-metadata-button]');
+            assert.dom('[data-test-edit-metadata-form]').doesNotExist();
+            await click('[data-test-edit-metadata-button]');
+
+            // Screenshot before changes
+            await percySnapshot(assert);
+            // Update Title
+            const editTitleTextArea = document.querySelectorAll('textarea')[0];
+            editTitleTextArea.textContent = 'A test title.';
+            assert.dom('[data-test-title-field]')
+                .containsText('A test title.', 'Metadata title field updates.');
+            // Update Description
+            const editDescriptionTextArea = document.querySelectorAll('textarea')[1];
+            editDescriptionTextArea.textContent = 'A test description.';
+            assert.dom('[data-test-title-field]')
+                .containsText('A test description.', 'Metadata description field updates.');
+            // Update Resource Type
+            await selectChoose('[data-test-select-resource-type]', 'InteractiveResource');
+            // Update Resource Language
+            await selectChoose('[data-test-select-resource-language]', 'InteractiveResource');
+            // Screenshot after changes
+            await percySnapshot(assert);
+            // Save changes
+            await click('[data-test-save-metadata-button]');
+        }
+        // Verify form closes
+        assert.dom('[data-test-edit-metadata-form]').doesNotExist();
+        assert.dom('[data-test-metadata-tab]').exists();
```

Comment:
You should verify all of the fields properly changed after the save, not just that the edit state changed. Similarly, you should verify that cancelling the edit changes the save state without changing them values of the fields.

Reply:
Technically, this is an expansion from the original ticket's scope. When the original ticket was made, we only had view functionality and not edit functionality, so we were just testing what we had at the time. I had forgotten about that over the course of time. I suppose we can pull this off onto a separate ticket.